.github/workflows/ci_pr.yml

```diff
@@ -2,7 +2,6 @@ name: Build and Test CI for Pull Requests
 
 on:
   pull_request:
-    branches: [ 'master', 'develop' ]
```

Comment:
This seems like a big change, could trigger a lot of extra CI actions that people don't expect. The more conservative way to do it would be to just add `ros2`, no?

The `workflow_dispatch` option is always there so you can trigger it manually for any branch.

Reply:
That was my initial thought too, but then I asked myself "will we ever want to open a pull request that doesn't have the build and test CI?". I do agree that it might trigger something extra that we don't want, I will revert this.